### PR TITLE
fix(permissioning): Fix restricted projects being visible

### LINKED ISSUE #19159: The project selector is showing projects that I have no access to.
## Bug description

The project selector is showing projects that I have no access to.

## How to reproduce

1. Login as https://posthoghelp.zendesk.com/agent/tickets/7723
2. Try to open a private project
3. No response (no toast, no error)

## Additional context

https://posthoghelp.zendesk.com/agent/tickets/7723

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Solves ZEN-7723. I discovered that if you were added to any private project, you were then able to see _all_ of them – even the ones you couldn't switch to because you didn't actually have the permissions.

## Changes

Fixed the `explicit_team_memberships` which was `OrganizationMembership -> ExplicitTeamMembership`, so effectively `Organization -> ExplicitTeamMembership` because `UserPermissions` are user-specific. This mapping must be `Team -> ExplicitTeamMembership` instead.

## How did you test this code?

Added some tests. The one that actually uncovered the issue is the `test_user_permissions.py` one.

### CODE REVIEW

File: ee/api/test/test_organization.py [hunk 1]
```diff
@@ -274,3 +274,20 @@ def test_feature_available_self_hosted_license_expired(self, patch_post):
...
             self.assertFalse(self.organization.is_feature_available("whatever"))
         License.PLANS = current_plans
+
+    def test_get_organization_restricted_teams_hidden(self):
+        self.organization_membership.level = OrganizationMembership.Level.MEMBER
+        self.organization_membership.save()
+        Team.objects.create(
+            organization=self.organization,
+            name="FORBIDDEN",
+            access_control=True,
```
Comment (pauldambra) [inline ee/api/test/test_organization.py:284]: I guess this is the line that forbids access?

Would be good to add a third team here where someone does have access so you can see both sides of the permissions setup

File: posthog/test/test_user_permissions.py [hunk 2]
```diff
@@ -45,6 +45,22 @@ def test_team_effective_membership_level_does_not_belong(self):
         with self.assertNumQueries(1):
             assert permissions.team(self.team).effective_membership_level is None
 
+    def test_team_effective_membership_level_membership_isolation(self):
```
Comment (pauldambra) [inline posthog/test/test_user_permissions.py:48]: ah, that's here 😊 

Should this be in the EE test file too? Or comments linking them?

Comment (pauldambra) [reply posthog/test/test_user_permissions.py:48]: Ah yeah ExplicitTeamMembership is an EE class so the tests should probably be there?

**Review by pauldambra — APPROVED** [review]
sorry, totally missed this!

some nit-picking on where tests live but otherwise LGTM

File: posthog/test/test_user_permissions.py (hunk 2 above)
Comment (Twixes) [reply posthog/test/test_user_permissions.py:48]: Hmm, perhaps. Though we've been fine with using these tests here for a while, so not gonna rock the boat for the time being.

**Pull request merged**